curriculum/challenges/english/25-front-end-development/lab-fcc-forum-leaderboard/673c91f0b934834bc4a3ecc2.md

```diff
@@ -0,0 +1,1446 @@
+1. Each image element should have an alt text with the value of the `name` property of the poster.
+1. Each image element should have a source with the value of the `avatar_template` property of the poster. In case `avatar_template` contains a relative path, you should set the source to `<avatarUrl>/<avatar_template>`.
+1. You should have a function named `showLatestPosts` that takes a single parameter.
+1. The `showLatestPosts` should extract the `users` and `topic_list` properties from the object passed as argument. Also, it should process the following properties of the `topics` object contained in `topic_list`:
+    - `id`: the id of the post
+    - `title`: the title of the post
+    - `views`: the number of views of the post 
+    - `posts_count`: the number of replies to the topic
+    - `slug`: the slug of the post
+    - `posters`: the posters for that topic
+    - `category_id`: an integer indicating the category id for the post
+    - `bumped_at`: a timestamp in the ISO 8601 format
+1. The `showLatestPosts` should set the inner HTML of `#posts-container` to a string made by joining `tr` elements, one for each item in `topics`.
+1. Each `tr` element should have five `td` elements in it:
+    - a `td` containing two anchor elements, one with the class of `post-title`, an `href` of `<forumTopicUrl><slug>/<id>`, an anchor text of `<title>`, and one obtained by calling `forumCategory` with `category_id`.
+    - a `td` containing a `div` element with class `avatar-container` that contains the images returned by the `avatars` function called with `posters` and `users` as arguments.
+    - a `td` containing the number of replies to the post. *Hint:* use `posts_count - 1`.
+    - a `td` containing the number of views of the post.
+    - a `td` containing the time passed since the last activity.
+1. You should have an async function named `fetchData`.
+1. The `fetchData` function should request data from `forumLatest` and call `showLatestPosts` passing it the response parsed as JSON.
+1. If there's an error when fetching data, the `fetchData` function should log the error to the console.
+
+# --hints--
+
+You should have a function named `timeAgo` that takes a single argument.
+
+```js
+assert.isFunction(timeAgo);
+assert.lengthOf(timeAgo, 1);
+```
+
+When the time difference between the time passed as argument and the current time is `50` minutes, `timeAgo` should return `50m ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * 60 * 50)).toISOString()
+}
+const expected = "50m ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `60` minutes and `10` seconds, `timeAgo` should return `1h ago` but found `--fcc-actual--`.
```

Comment:
This works to check if someone is using `ceil` probably. To check `round` vs `floor` it probably need 115 minutes that should give 1h